### ISSUE TITLE
dice_loss cannot converge

### ISSUE THREAD

**Issue by Siannodel** [KangBK0120/PERT#4]
I have a problem, dice_loss cannot converge.
The final trained model has an erasure effect, but the mask_out is completely wrong, and then I output the dice_loss during the training process and found that it does not drop at all.

**Comment by Siannodel** [comment]
I think I know what the problem is：
`class TLN(nn.Module):
    def __init__(self):
        super(TLN, self).__init__()

        self.pspmodule = PSPModule(256, 512)
        self.conv1 = ConvBlock(
            in_channels=512, out_channels=256, kernel_size=3, stride=1, padding=1
        )
        self.conv2 = ConvBlock(
            in_channels=256, out_channels=256, kernel_size=3, stride=1, padding=1
        )
        self.conv3 = ConvBlock(
            in_channels=512, out_channels=128, kernel_size=3, stride=1, padding=1
        )
        self.conv4 = ConvBlock(in_channels=256, out_channels=32, kernel_size=3, stride=1, padding=1)
        self.conv5 = ConvBlock(in_channels=32, out_channels=1, kernel_size=1, stride=1, padding=0)
        self.relu = nn.ReLU()

    def forward(
        self, feat: torch.Tensor, res2_feat: torch.Tensor, res4_feat: torch.Tensor
    ) -> torch.Tensor:
        out = self.pspmodule(feat)
        out = self.conv1(out)
        out = F.interpolate(out, scale_factor=2, mode="bilinear")
        out = self.conv2(out)

        out = torch.cat([out, res4_feat], dim=1)
        out = F.interpolate(out, scale_factor=2, mode="bilinear")
        out = self.conv3(out)

        out = torch.cat([out, res2_feat], dim=1)
        out = F.interpolate(out, scale_factor=2, mode="bilinear")
        out = self.conv4(out)
        out = self.conv5(out)

        out = F.interpolate(out, scale_factor=4, mode="bilinear")
        mask = F.sigmoid(out)

        return mask`

self.conv5() has relu，so all outputs >= 0, then sigmoid's output >= 0.5

**Comment by KangBK0120** [comment]
Hi, 

thanks for your concise comment. I will check and update the codes.

Have you tried any training with removing ReLU in conv5?

**Comment by KangBK0120** [comment]
Hi,

I fixed some bugs in the model codes. 
However, the performance is not that improved from the past.

I updated the model weight with the debugged one.
